samples/snippets/cloud-client/src/test/java/privateca/SnippetsIT.java

```diff
@@ -138,73 +127,40 @@ public static void cleanUp() throws InterruptedException, ExecutionException, IO
-  public static void createAsymmetricSignKey() throws IOException {
-    // Initialize client that will be used to send requests. This client only
-    // needs to be created once, and can be reused for multiple requests. After
-    // completing all of your requests, call the "close" method on the client to
-    // safely clean up any remaining background resources.
-    try (KeyManagementServiceClient client = KeyManagementServiceClient.create()) {
-      // Build the parent name from the project, location, and key ring.
-      KeyRingName keyRingName = KeyRingName.of(PROJECT_ID, KMS_LOCATION, KEY_RING_ID);
-
-      // Build the asymmetric key to create.
-      CryptoKey key =
-          CryptoKey.newBuilder()
-              .setPurpose(CryptoKeyPurpose.ASYMMETRIC_SIGN)
-              .setVersionTemplate(
-                  CryptoKeyVersionTemplate.newBuilder()
-                      .setAlgorithm(CryptoKeyVersionAlgorithm.RSA_SIGN_PKCS1_2048_SHA256))
-              .build();
-
-      // Create the key.
-      CryptoKey createdKey = client.createCryptoKey(keyRingName, KEY_ID, key);
-      System.out.printf("Created asymmetric key: %s%n", createdKey.getName());
-    }
-  }
-
-  public static void cleanupCertificateSignKey() throws IOException, InterruptedException {
-    try (KeyManagementServiceClient client = KeyManagementServiceClient.create()) {
-      CryptoKeyVersionName cryptoKeyVersionName =
-          CryptoKeyVersionName.of(PROJECT_ID, KMS_LOCATION, KEY_RING_ID, KEY_ID, VERSION_ID);
-      // Destroy the crypto key version.
-      CryptoKeyVersion cryptoKeyVersion = client.destroyCryptoKeyVersion(cryptoKeyVersionName);
-      sleep(5);
-      // If the response has destroy time, then the version is successfully destroyed.
-      Assert.assertTrue(cryptoKeyVersion.hasDestroyTime());
-    }
-  }
-
   // Wait for the specified amount of time.
   public static void sleep(int seconds) throws InterruptedException {
     TimeUnit.SECONDS.sleep(seconds);
   }
 
+  // Create an asymmetric key pair to be used in certificate signing.
+  public static ByteString createPublicKey()
+      throws NoSuchAlgorithmException, NoSuchProviderException, IOException {
+    Security.addProvider(new BouncyCastleProvider());
+    System.out.println("BouncyCastle provider added.");
+
+    KeyPairGenerator generator = KeyPairGenerator.getInstance("RSA", "BC");
+    generator.initialize(KEY_SIZE);
+    KeyPair keyPair = generator.generateKeyPair();
+    System.out.println("RSA key pair generated.");
+
+    // Only the public key will be used to create the certificate.
+    // Save the private key in your machine to verify the certificate.
+    RSAPrivateKey privateKey = (RSAPrivateKey) keyPair.getPrivate();
+    RSAPublicKey publicKey = (RSAPublicKey) keyPair.getPublic();
+
+    // Construct the PemObject for public key.
+    PemObject pemObject = new PemObject("RSA PUBLIC KEY", publicKey.getEncoded());
```

Comment:
It's worth double checking whether the encoded public key will be in SPKI or PKCS#1 format. If SPKI, then I believe the most appropriate header here will be "PUBLIC KEY" rather than "RSA PUBLIC KEY". I couldn't easily confirm this with a quick search, so it might be worth exporting this to a file and using openssl to check the actual format.

https://datatracker.ietf.org/doc/html/rfc7468#section-13

See also the answers to this question: https://stackoverflow.com/q/18039401

Reply:
Thank you Peter for this great insight!
Bouncy Castle framework allows us the flexibility to choose the encoding format, be it PKCS#1 or SPKI etc., 

As per your suggestion, I tried encoding it in SPKI format, verified with openssl and also went ahead to create & check the validity of the issued cert and it works fine. Will update it in the following commit.